static/ankersrv.js

```diff
@@ -50,12 +50,107 @@ $(function () {
         }, +alert_timeout);
     });
 
+    /**
+     * Get temperature from input
+     * @param {number} temp Temperature in Celsius
+     * @returns {number} Rounded temperature
+     */
+    function getTemp(temp) {
+        return Math.round(temp / 100);
+    }
+
+    /**
+     * Calculate the percentage between two numbers
+     * @param {number} layer
+     * @param {number} total
+     * @returns {number} percentage
+     */
+    function getPercentage(progress) {
+        return Math.round(((progress / 100) * 100) / 100);
+    }
+
+    /**
+     * Truncate a string to a specific length
+     * @param {string} str
+     * @param {number} maxLength
+     * @returns {string} Truncated string
+     */
+    function truncStr(str, maxLength) {
+        if (str.length > maxLength) {
+            return str.slice(0, maxLength) + "...";
+        } else {
+            return str;
+        }
+    }
+
+    /**
+     * Convert time in seconds to hours, minutes, and seconds format
+     * @param {number} seconds
+     * @returns {string} Formatted time string
+     */
+    function getTime(seconds) {
+        const hours = Math.floor(seconds / 3600);
+        const minutes = Math.floor((seconds % 3600) / 60);
+        const remainingSeconds = seconds % 60;
+
+        const timeString = `${hours.toString().padStart(2, "0")}:${minutes
+            .toString()
+            .padStart(2, "0")}:${remainingSeconds.toString().padStart(2, "0")}`;
+
+        return timeString;
+    }
+
+    /**
+     * Get the filament weight based on its length
+     * @param {number} length
+     * @returns {number} Weight
+     */
+    function getWeight(length) {
+        const diameter = 1.75;
+        const density = 1.24;
```

Comment:
This function actually needs to go away. I was originally using it to convert how much Filament was being used because the print I was doing was returning `mm` for the filament measurement but it didn't seem to be accurate so I stripped it from the visible fields.